window-purpose-core.el

```diff
@@ -212,6 +212,9 @@ If no purpose was determined, return `default-purpose'."
      (purpose--buffer-purpose-mode buffer-or-name
                                    purpose--default-mode-purposes)))
 
+   ;; If the buffer is visiting a file, fallback to 'edit purpose
+   (when (buffer-file-name (get-buffer buffer-or-name))
+     'edit)
```

Comment:
Should use a new custom `default-file-purpose` variable, which has a default value of `edit`.